lib/mayaUsd/ufe/UsdLight.cpp

```diff
@@ -0,0 +1,499 @@
+class SetValueUndoableCommandImpl : public Ufe::SetValueUndoableCommand<ValueTypeIn>
+{
+public:
+    using ValueTypeNonRef = typename std::remove_reference<ValueTypeIn>::type;
+    using ValueTypeOut = typename std::remove_const<ValueTypeNonRef>::type;
+
+    typedef std::function<ValueTypeOut(const UsdPrim& prim)>      GetterFunc;
+    typedef std::function<void(const UsdPrim& prim, ValueTypeIn)> SetterFunc;
+
+    SetValueUndoableCommandImpl(const Ufe::Path& path, const GetterFunc& gf, const SetterFunc& sf)
+        : Ufe::SetValueUndoableCommand<ValueTypeIn>(path)
+        , getterFunc(gf)
+        , setterFunc(sf)
+    {
+    }
+
+    ~SetValueUndoableCommandImpl() override { }
+
+    bool set(ValueTypeIn v) override
+    {
+        redoValue = v;
+        return true;
+    }
+
+    void execute() override
+    {
+        if (auto pItem
+            = std::dynamic_pointer_cast<UsdSceneItem>(Ufe::BaseUndoableCommand::sceneItem())) {
+            undoValue = getterFunc(pItem->prim());
+            setterFunc(pItem->prim(), redoValue);
+        }
+    }
+
+    void undo() override
+    {
+        if (auto pItem
+            = std::dynamic_pointer_cast<UsdSceneItem>(Ufe::BaseUndoableCommand::sceneItem())) {
+            setterFunc(pItem->prim(), undoValue);
+        }
+    }
+
+    void redo() override
+    {
+        if (auto pItem
+            = std::dynamic_pointer_cast<UsdSceneItem>(Ufe::BaseUndoableCommand::sceneItem())) {
+            setterFunc(pItem->prim(), redoValue);
+        }
+    }
+
+private:
+    GetterFunc getterFunc;
+    SetterFunc setterFunc;
+
+    ValueTypeOut redoValue;
+    ValueTypeOut undoValue;
+};
+
+UsdLight::UsdLight()
+    : Light()
+{
+}
+
+UsdLight::UsdLight(const UsdSceneItem::Ptr& item)
+    : Light()
+    , fItem(item)
+{
+}
+
+UsdLight::Ptr UsdLight::create(const UsdSceneItem::Ptr& item)
+{
+    return std::make_shared<UsdLight>(item);
+}
+
+#if PXR_VERSION < 2111
+using UsdLuxLightCommon = UsdLuxLight;
+#else
+using UsdLuxLightCommon = UsdLuxLightAPI;
+#endif
+
+//------------------------------------------------------------------------------
+// Ufe::Light overrides
+//------------------------------------------------------------------------------
+
+const Ufe::Path& UsdLight::path() const { return fItem->path(); }
+
+Ufe::SceneItem::Ptr UsdLight::sceneItem() const { return fItem; }
+
+Ufe::Light::Type UsdLight::type() const
+{
+    const UsdPrim usdPrim = prim();
+
+    if (usdPrim.IsA<UsdLuxDistantLight>()) {
+        return Ufe::Light::Directional;
+    } else if (usdPrim.IsA<UsdLuxRectLight>()) {
+        return Ufe::Light::Area;
+    } else if (usdPrim.IsA<UsdLuxSphereLight>()) {
+        const UsdLuxShapingAPI shapingAPI(usdPrim);
+        return shapingAPI.GetShapingConeAngleAttr().IsValid() ? Ufe::Light::Spot
+                                                              : Ufe::Light::Point;
+    }
+
+    return Ufe::Light::Invalid;
+}
+
+float getLightIntensity(const UsdPrim& prim)
```

Comment:
Is it useful for these free functions to be visible in the MayaUsd::ufe namespace?  Or should they be private in the unnamed namespace for this translation unit?  No preference either way, just wondering.